src/extension.ts

```diff
@@ -207,6 +207,20 @@ export async function activate(context: ExtensionContext) {
       console.log(error);      
     }
   });
+  // extension connector
+  commands.registerCommand('rmxPlugins.deploy', async () => {
```

Comment:
While I am not really against adding a `deploy` command here, 
I think `ethcode` should listen on compilation result.
e.g it has either a remix plugin layer that will allow to listen using the remix API, or (that would be ok for the moment, `ethcodeApi.loadCompiled(compilationResult);` can be called each time there's a compilation result coming from remix-plugin).
And I suppose this will be replace later on by a direct connection with Juan Blanco's extension

Reply:
There are 2 different compilers inside remix vscode extension. In future there could be more. And also there is vyper compiler.
We can try to load the compilation results directly from other compiler extensions. I would also love to load it from remix vscode extension.

Yes your idea of having a plugin (may be just another command plugin) to load the compilation results and then call ethcode can be the exact thing we want. Let me try and I will get back to you. :crossed_fingers: